test/integration/lib/render.js

```diff
@@ -147,16 +148,17 @@ async function setupLayout(options) {
     }
 }
 
-async function getExpectedImages(currentTestName) {
-    const currentFixture = fixtures[currentTestName];
-
+async function getExpectedImages(currentTestName, currentFixture) {
     // there may be multiple expected images, covering different platforms
     const expectedPaths = [];
     for (const prop in currentFixture) {
         if (prop.indexOf('expected') > -1) {
-            let path = `${currentTestName}/${prop}.png`;
-            // regression tests with # in the name need to be sanitized
-            path = encodeURIComponent(path);
+            const path = `/${currentTestName}/${prop}.png`
+                .split('/')
+                // regression tests with # in the name need to be sanitized
+                .map(p => encodeURIComponent(p))
+                .join('/');
```

Comment:
@stepankuzmin Could you share context for this change? Was there a performance issue caused by encoding full paths?

Reply:
When we encode the whole URL path with `encodeURIComponent`, it's being served with the `testem`'s built-in server and not our `serve-static` middleware. It might be okay since it's unnecessary to cache the `expectend.png` in the browser, but I think it's better to serve it with our server to be consistent with the other assets.

https://github.com/mapbox/mapbox-gl-js/blob/c861dec0d4b2b78734881a4d6e64baf5d9b54892/test/integration/lib/middlewares.cjs#L22-L24